cli/tests/unit/permissions_test.ts

```diff
@@ -19,8 +20,33 @@ unitTest(async function permissionNetInvalidHost(): Promise<void> {
   }, URIError);
 });
 
+unitTest(async function permissionQueryReturnsEventTarget() {
+  const status = await Deno.permissions.query({ name: "hrtime" });
+  assert(["granted", "denied", "prompt"].includes(status.state));
+  let called = false;
+  status.addEventListener("change", () => {
+    called = true;
+  });
+  status.dispatchEvent(new Event("change"));
+  assert(called);
+  assert(status === (await Deno.permissions.query({ name: "hrtime" })));
```

Comment:
Is there an integration test that check event listeners after actual granting of permissions? If not would be great to have one, but I get that it might be hard to do (given that prompt tests are rather flaky)

Reply:
No there isn't, but in a way I think we should use integration tests more sparingly and not have them be a coverage for lack of unit tests. (And we need another flaky test like we need a hole in our heads)